tests/run-build.sh

```diff
@@ -38,8 +38,10 @@ if [ "${SEMAPHORE-}" == true ] ; then
         fi
 fi
 
-if [ ${SRC_CHANGES} -eq 0 ] && [ ${DOC_CHANGES} -eq 0 ]; then
-    echo No changes detected.
+HEAD=`git rev-parse HEAD`
+MASTER=`git rev-parse origin/master`
+if [[ ${HEAD} != ${MASTER} ]] && [[ ${SRC_CHANGES} -eq 0 && ${DOC_CHANGES} -eq 0 ]]; then
```

Comment:
Even if we don't skip running the tests on master here, we'll skip them later because we still check `SRC_CHANGES`, don't we?